### PR TITLE
Compose wrapper

### PR DESCRIPTION
**Pull request by chuckbutler**
Adds a docker-compose lib stub (early iteration shipping w/ the charm)

This will eventually move to its own pypi module. For now, exploratory dev shipping w/ the docker charm.

This includes a wrapper class to reduce overall code complexity when consuming docker-compose based formations. You can use this class like the following example code:

    from charms.docker.compose include Compose
    c = Compose('files/mycluster')
    c.start_service()  # starts all services in docker-compose.yml
    c.stop_service('nginx')  # stops nginx listed in docker-compose.yml

Arguments to the Compose Initializer:

workspace - path to find the docker-compose.yaml or docker-compose.yml

This class is a light weight wrapper, and subject to change.

### CODE REVIEW

File: lib/charms/docker/compose.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from shlex import split
+from subprocess import check_call
+
+# Wrapper and convenience methods for charming w/ docker compose in python
+class Compose:
+    def __init__(self, workspace):
+        self.workspace = workspace
+        self.validate_workspace()
+
+    def start_service(self, service=None):
```
Comment (mbruzek) [inline lib/charms/docker/compose.py]: we should name these methods similar to the actual docker-compose command.
Suggestion:  def up(self, service=None):

File: lib/charms/docker/compose.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        if service:
+            cmd = "docker-compose up -d {}".format(service)
+        else:
+            cmd = "docker-compose up -d"
+        self.run(cmd)
+
+    def kill_service(self, service=None):
```
Comment (mbruzek) [inline lib/charms/docker/compose.py]: suggestion: def kill(self, service=None):

File: lib/charms/docker/compose.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        if service:
+            cmd = "docker-compose kill -d {}".format(service)
+        else:
+            cmd = "docker-compose kill"
+        self.run(cmd)
+
+    def run(self, cmd):
```
Comment (mbruzek) [inline lib/charms/docker/compose.py]: doc strings so we know that "run" changes dir to run the command.

File: lib/charms/docker/compose.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        with chdir(self.workspace):
+            check_call(split(cmd))
+
+
+    def validate_workspace(self):
```
Comment (mbruzek) [inline lib/charms/docker/compose.py]: doc string all the things!

**Comment by mbruzek** [comment]
+1 LGTM  - Had some suggestions for improvement but only suggestions.  They should not block merging if you disagree or want a different mp to address.

**Comment by chuckbutler** [comment]
Superseded by #19

**Pull request closed without merging**